com.github.ransome1.sleek.yml

```diff
@@ -0,0 +1,48 @@
+app-id: com.github.ransome1.sleek
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+branch: stable
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  #- --socket=pulseaudio
+  - --share=network
+#build-options:
+#  env:
+#    NPM_CONFIG_LOGLEVEL: info
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node12
+modules:
+  - name: yarn
+    buildsystem: simple
+    build-commands:
+      - 'cp -a * /app'
+    # Only used for building, so clean it up afterwards.
+    cleanup:
+      - '*'
+    sources:
+      - type: archive
+        url: https://github.com/yarnpkg/yarn/releases/download/v1.19.1/yarn-v1.19.1.tar.gz
+        sha256: 34293da6266f2aae9690d59c2d764056053ff7eebc56b80b8df05010c3da9343
+
+  - name: sleek
+    build-options:
+      append-path: /usr/lib/sdk/node12/bin
+      env:
+        ELECTRON_CACHE: /run/build/sleek/flatpak-node/electron-cache
+        npm_config_nodedir: /usr/lib/sdk/node12
+    buildsystem: simple
+    build-commands:
+      - npm install --prefix=sleek --offline --cache=$FLATPAK_BUILDER_BUILDDIR/flatpak-node/npm-cache
+      - . flatpak-node/electron-builder-arch-args.sh; yarn run --offline dist -- $ELECTRON_BUILDER_ARCH_ARGS
```

Comment:
I think this is the problem, this is using `npm` to download the dependencies but `yarn` to run the build. (This isn't really something supported outside Flatpak either afaik, you'll run into weird lockfile issues.) If the app uses yarn, follow the `yarn install` command in the flatpak-node-generator examples, and run it as `flatpak-node-generator yarn ...`.

Reply:
Good catch; the electron failure happens on `npm install` step though. Not sure why, it's there in generated-sources.